c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/SystemInfo.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Apache NiFi - MiNiFi
+ * Copyright 2014-2018 The Apache Software Foundation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.api;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import javax.validation.constraints.DecimalMax;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Size;
+import java.io.Serializable;
+import java.util.Objects;
+
+@ApiModel
+public class SystemInfo implements Serializable {
+    private static final long serialVersionUID = 830834593998474779L;
+
+    public static final int MACHINE_ARCH_MAX_SIZE = 100;
+    public static final int OS_MAX_SIZE = 100;
+
+    @Size(max = MACHINE_ARCH_MAX_SIZE)
+    @ApiModelProperty("Machine architecture of the device, e.g., ARM, x86")
+    private String machineArch;
+
+    @Size(max = OS_MAX_SIZE)
+    private String operatingSystem;
+
+    @Min(0)
+    @Max(Long.MAX_VALUE)
+    @ApiModelProperty(value = "Size of physical memory of the device in bytes", allowableValues = "range[0, 9223372036854775807]")
+    private Long physicalMem;
+
+    @Min(0)
+    @Max(Integer.MAX_VALUE)
+    @ApiModelProperty(
+            value = "Number of virtual cores on the device",
+            name = "vCores",
+            allowableValues = "range[0, 2147483647]")
+    private Integer vCores;
```

Comment:
As mentioned elsewhere, the allowableValues seems to imply that this cannot be null.

Reply:
Good catch, also leftover from JAXB annotations, will change to primitives

Reply:
I don't feel strongly, but I would have a slight preference to leave optional fields as nullable and update the allowable values to reflect that. In this context, `null == unknown`. Personally, I find this easier to deal with than inferring if 0 is a default value or explicitly passed by the client, when debugging, etc.

Reply:
@exceptionfactory What are your thoughts on this?

Reply:
Leaving it a nullable type makes sense if that is the intention. If the allowable values is informational, then this is fine as it stands.